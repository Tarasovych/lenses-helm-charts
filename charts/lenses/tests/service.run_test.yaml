suite: test service resource
templates:
  - service.yaml
tests:
  - it: should have ClusterIP type by default
    set:
      service:
        enabled: true
    asserts:
      # Check the rendered manifest has not changed
      # Review the changes and if they are verified use `helm unittest -u .` to update the manifest
      # Writing tests that cover this chaneg is highly recommended
      - matchSnapshot: {}
      - equal:
          path: spec.type
          value: ClusterIP
  - it: should have NodePort type if specified
    set:
      service:
        enabled: true
        type: NodePort
    asserts:
      - equal:
          path: spec.type
          value: NodePort
  - it: should have custom annotations if specified
    set:
      service:
        enabled: true
        type: NodePort
        annotations:
          foo: bar
    asserts:
      - equal:
          path: spec.type
          value: NodePort
      - equal:
          path: metadata.annotations.foo
          value: bar

  - it: should have custom externalTrafficPolicy if specified
    set:
      service:
        enabled: true
        type: NodePort
        externalTrafficPolicy: Cluster
    asserts:
      - equal:
          path: spec.type
          value: NodePort
      - equal:
          path: spec.externalTrafficPolicy
          value: Cluster
