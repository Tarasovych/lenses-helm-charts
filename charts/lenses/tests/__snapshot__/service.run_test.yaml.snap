should have ClusterIP type by default:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.3.6
        component: lenses
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-service
        release: RELEASE-NAME
      name: RELEASE-NAME
    spec:
      ports:
      - name: lenses
        port: 80
        targetPort: 3030
      - name: http-metrics
        port: 9102
      selector:
        app: RELEASE-NAME
        component: lenses
      type: ClusterIP
